cmd/kubeadm/app/apis/kubeadm/v1alpha3/doc.go

```diff
@@ -19,127 +19,255 @@ limitations under the License.
-//	  hostPath: "/etc/some-path"
-//	  mountPath: "/etc/some-pod-path"
-//	  writable: true
-//	  pathType: File
-//	schedulerExtraVolumes:
-//	- name: "some-volume"
-//	  hostPath: "/etc/some-path"
-//	  mountPath: "/etc/some-pod-path"
-//	  writable: true
-//	  pathType: File
-//	apiServerCertSANs:
-//	- "10.100.1.1"
-//	- "ec2-10-100-0-1.compute-1.amazonaws.com"
-//	certificatesDir: "/etc/kubernetes/pki"
-//	imageRepository: "k8s.gcr.io"
-//	unifiedControlPlaneImage: "k8s.gcr.io/controlplane:v1.12.0"
-//	auditPolicy:
-//	  # https://kubernetes.io/docs/tasks/debug-application-cluster/audit/#audit-policy
-//	  path: "/var/log/audit/audit.json"
-//	  logDir: "/var/log/audit"
-//	  logMaxAge: 7 # in days
-//	featureGates:
-//	  selfhosting: false
-//	clusterName: "example-cluster"
-//
-// TODO: The BootstrapTokenString object should move out to either k8s.io/client-go or k8s.io/api in the future
-// (probably as part of Bootstrap Tokens going GA). It should not be staged under the kubeadm API as it is now.
+// Package v1alpha3 defines the v1alpha3 version of the kubeadm config file format, that is a big step
+// forward the objective of graduate kubeadm config to beta.
 //
+// One of the biggest changes introduced by this release is the re-design of how component config
+// can be provided to kubeadm; this will enable a improved stability of the kubeadm config while the efforts for
+// the implementation of component config across Kubernetes ecosystem continues.
+//
+// Another important change is the separation between cluster wide setting and runtime or node specific
+// settings, that is functional to the objective to introduce support for HA clusters in kubeadm.
+//
+// Migration from old kubeadm config versions
+//
+// Please convert your v1alpha2 configuration files to v1alpha3 using the kubeadm config migrate command of kubeadm v1.12.x
+// (conversion from older releases of kubeadm config files requires older release of kubeadm as well e.g. v1alpha1, v1.11.x).
+//
+// Nevertheles, kubeadm v1.12.x will support reading from v1alpha2 version of the kubeadm config file format, but this support
+// will be dropped in v1.13 release.
+//
+// Basics
+//
+// The preferred way to configure kubeadm is to pass YAML a config file with the --config option: some of the
+// configuration options defined in the kubeadm config file are also available as command line flags, but only
+// the most common/simple use case are supported with this approach.
+//
+// A kubeadm config file could contain multiple configuration types separated using three dashes (“---”).
+//
+// The kubeadm config print-defaults command print the default values for all the kubeadm supported configuration types.
+//
+//     apiVersion: kubeadm.k8s.io/v1alpha3
+//     kind: InitConfiguration
+//         ...
+//     ---
+//     apiVersion: kubeadm.k8s.io/v1alpha3
+//     kind: ClusterConfiguration
+//         ...
+//     ---
+//    apiVersion: kubelet.config.k8s.io/v1beta1
```

Comment:
Spacing is lost here between this line and the one above. This can be a bit confusing to the reader if he or she has no more context that this.